hydrolib/core/basemodel.py

```diff
@@ -22,7 +31,35 @@ class Config:
         use_enum_values = True
         extra = "forbid"  # will throw errors so we can fix our models
         allow_population_by_field_name = True
-        alias_generator = to_lowercase
+        alias_generator = to_key
+
+    def is_file_link(self) -> bool:
+        """Generic attribute for models backed by a file."""
+        return False
+
+    def is_intermediate_link(self) -> bool:
+        """Generic attribute for models that have children fields that could contain files."""
+        return self.is_file_link()
```

Comment:
Could this be determined dynamically, or is that overkill?

Reply:
Could be, we're now mostly looking for a (child)instance of `FileModel`. Then again, that's unknown to check for on the `BaseModel`, so I thought this was a nice way of doing it and could provide a way to also stop recursively parsing eventually.